pkg/agent/agent.go

```diff
@@ -87,6 +92,23 @@ func (a *Agent) readPump() {
 		return a.conn.SetReadDeadline(time.Now().Add(pongWait))
 	})
 
+	a.conn.SetCloseHandler(func(err int, msg string) error {
+		a.logger.Debug("peer disconnected", logging.Fields{"room": a.room, "peer": a.peer.UID, "error_code": err, "error_msg": msg})
+		
+		jsonMessage, jsonError := json.Marshal(DisconnectPayload{"peer_disconnected", a.peer.UID})
+		if jsonError != nil {
+			a.logger.Error("can't marshal object to json")
```

Comment:
This should be more specific. We can also add fields to the message so we know for which peer it happened.